### ISSUE TITLE
False positive error with bidirectional type inference for call expression

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#8463]
The following should not result in a type error but does currently:

```python
from typing import Iterable, TypeVar

T = TypeVar("T")

class Item:
    ...

class Function(Item):
    ...

def genfunctions() -> Iterable[Function]:
    return []

def make_list(val: Iterable[T]) -> list[T]:
    return list(val)

x: Item | list[Item | None] = make_list(genfunctions())
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.373.
